homeassistant/components/seventeentrack/config_flow.py

```diff
@@ -0,0 +1,124 @@
+"""Adds config flow for 17track.net."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from py17track import Client as SeventeenTrackClient
+from py17track.errors import SeventeenTrackError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.schema_config_entry_flow import (
+    SchemaFlowFormStep,
+    SchemaOptionsFlowHandler,
+)
+
+from .const import (
+    CONF_SHOW_ARCHIVED,
+    CONF_SHOW_DELIVERED,
+    DEFAULT_SHOW_ARCHIVED,
+    DEFAULT_SHOW_DELIVERED,
+    DOMAIN,
+)
+
+CONF_SHOW = {
+    vol.Optional(CONF_SHOW_ARCHIVED, default=DEFAULT_SHOW_ARCHIVED): bool,
+    vol.Optional(CONF_SHOW_DELIVERED, default=DEFAULT_SHOW_DELIVERED): bool,
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+OPTIONS_SCHEMA = vol.Schema(CONF_SHOW)
+OPTIONS_FLOW = {
+    "init": SchemaFlowFormStep(OPTIONS_SCHEMA),
+}
+
+USER_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class SeventeenTrackConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """17track config flow."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> SchemaOptionsFlowHandler:
+        """Get options flow for this handler."""
+        return SchemaOptionsFlowHandler(config_entry, OPTIONS_FLOW)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+
+        errors = {}
+        if user_input:
+            client = await self._get_client()
+
+            try:
+                if not await client.profile.login(
+                    user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+                ):
+                    _LOGGER.error("Invalid username and password provided")
```

Comment:
This error doesn't add anything as the error message already reflects this

Reply:
removed